### DISCUSSION TITLE
Can not display lsp diagnostics msg with newline

### DISCUSSION THREAD

**Discussion by holicc** [NvChad/NvChad#720] [Q&A] [answered]
can not say all the error msg from lsp :

![image](https://user-images.githubusercontent.com/19146591/148652044-6b4b1887-a905-47f2-9ae8-fab81d3f46f9.png)

**Comment by siduck** [comment, marked as answer]
> can not say all the error msg from lsp :
> 
> ![image](https://user-images.githubusercontent.com/19146591/148652044-6b4b1887-a905-47f2-9ae8-fab81d3f46f9.png)

which language is this? and which lsp

**Reply by holicc** [reply]
> > can not say all the error msg from lsp :
> > ![image](https://user-images.githubusercontent.com/19146591/148652044-6b4b1887-a905-47f2-9ae8-fab81d3f46f9.png)
> 
> which language is this? and which lsp

language: **golang** , lsp: **gopls**

**Reply by holicc** [reply]
is there anyway display diagnostics msg with pop window , like lsp hover ?

**Reply by siduck** [reply]
```
vim.lsp.handlers["textDocument/publishDiagnostics"] = vim.lsp.with(vim.lsp.diagnostic.on_publish_diagnostics, {
   virtual_text = false,
   underline = true,
   signs = true,
})

**Reply by siduck** [reply]
to show the diagnostics in floating , you need to press ge every time on the underlined error 

so add some autocmds for it 

```
vim.cmd "autocmd CursorHold * lua vim.diagnostic.open_float()"
vim.cmd "autocmd CursorHoldI * silent! lua vim.lsp.buf.signature_help()"
```

**Reply by siduck** [reply]
so if the cursor's on the underlined error , then a small popup will open up 

![image](https://user-images.githubusercontent.com/59060246/148668584-f9840703-3229-450e-9123-88f75decfa9c.png)

**Reply by holicc** [reply]
how to make lsp `lua vim.lsp.buf.implementation()` in floating ?

this not work . i'm not good at lua and nvim 🙏

```
vim.lsp.handlers["textDocument/implementation"] = vim.lsp.with(vim.lsp.implementation, {
      virtual_text = false,
      underline = true,
      signs = true,
   })
```

**Reply by siduck** [reply]
@holicc my bad! add the code in your custom lspconfig , for example 
![image](https://user-images.githubusercontent.com/59060246/148722063-87162e30-ad9b-45ab-acc9-578d75ddc3e9.png)


also you seem new to lua and nvim , please check our docs from starting

**Reply by holicc** [reply]
sorry my english 😂 ,  `diagnostics` works great ! 

this is another question .

when i press `gi` i want to make lsp  `implementation` msg in folating ,  like diagnostics does.

> how to make lsp `lua vim.lsp.buf.implementation()` in floating ?
> 
> this not work . i'm not good at lua and nvim 🙏
> 
> ```
> vim.lsp.handlers["textDocument/implementation"] = vim.lsp.with(vim.lsp.implementation, {
>       virtual_text = false,
>       underline = true,
>       signs = true,
>    })
> ```

**Reply by siduck** [reply]
> sorry my english joy , `diagnostics` works great !
> 
> this is another question .
> 
> when i press `gi` i want to make lsp `implementation` msg in folating , like diagnostics does.
> 
> > how to make lsp `lua vim.lsp.buf.implementation()` in floating ?
> > this not work . i'm not good at lua and nvim pray
> > ```
> > vim.lsp.handlers["textDocument/implementation"] = vim.lsp.with(vim.lsp.implementation, {
> >       virtual_text = false,
> >       underline = true,
> >       signs = true,
> >    })
> > ```

lsp implementation? whats that :l

**Reply by holicc** [reply]
> > sorry my english joy , `diagnostics` works great !
> > this is another question .
> > when i press `gi` i want to make lsp `implementation` msg in folating , like diagnostics does.
> > > how to make lsp `lua vim.lsp.buf.implementation()` in floating ?
> > > this not work . i'm not good at lua and nvim pray
> > > ```
> > > vim.lsp.handlers["textDocument/implementation"] = vim.lsp.with(vim.lsp.implementation, {
> > >       virtual_text = false,
> > >       underline = true,
> > >       signs = true,
> > >    })
> > > ```
> 
> lsp implementation? whats that :l

![image](https://user-images.githubusercontent.com/19146591/148727863-73566ace-ea19-4d9b-82c7-d7d216bdb717.png)

**Reply by holicc** [reply]
when i press `gi` open a window blow show code implementions : 
![image](https://user-images.githubusercontent.com/19146591/148728874-87b0ba92-a4b1-4d31-b7f0-df642e38fc64.png)
how to make it in floating ?

**Reply by siduck** [reply]
idk that, ask it in lspconfig repo
